backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/export/SVGExporter.java

```diff
@@ -0,0 +1,273 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.diagrams.export;
+
+import java.awt.Color;
+import java.text.MessageFormat;
+
+import org.eclipse.sirius.web.diagrams.Diagram;
+import org.eclipse.sirius.web.diagrams.Edge;
+import org.eclipse.sirius.web.diagrams.EdgeStyle;
+import org.eclipse.sirius.web.diagrams.INodeStyle;
+import org.eclipse.sirius.web.diagrams.ImageNodeStyle;
+import org.eclipse.sirius.web.diagrams.Label;
+import org.eclipse.sirius.web.diagrams.LabelStyle;
+import org.eclipse.sirius.web.diagrams.Node;
+import org.eclipse.sirius.web.diagrams.Position;
+import org.eclipse.sirius.web.diagrams.RectangularNodeStyle;
+import org.eclipse.sirius.web.diagrams.Size;
+
+/**
+ * Service used to export a diagram to SVG.
+ *
+ * @author rpage
+ */
+public class SVGExporter {
+    private static final String OPEN_CLASS_ATTRIBUTE = "class=\""; //$NON-NLS-1$
+
+    private static final String OPEN_G = "<g "; //$NON-NLS-1$
+
+    private static final String SEMI_COLON = "; "; //$NON-NLS-1$
+
+    private static final String COMMA = ", "; //$NON-NLS-1$
+
+    private static final String END_G_ELEMENT = "</g>"; //$NON-NLS-1$
+
+    private static final String END_QUOTE = "\" "; //$NON-NLS-1$
+
+    private static final String CLOSE = ">"; //$NON-NLS-1$
```

Comment:
Use something like `@SuppressWarnings("checkstyle:MultipleStringLiterals")` on this class to deactivate just the check style rule for multiple occurrences of a same string in this specific file. It creates more readability issues in this specific case.